src/services/pagoPAClientFactory.ts

```diff
@@ -2,17 +2,25 @@
  * This service builds API clients.
  */
 
-import { IPagoPAClientFactoryInterface } from "./IPagoPAClientFactory";
+import {
+  IGetClientParams,
+  IPagoPAClientFactoryInterface
+} from "./IPagoPAClientFactory";
 
 import { PagoPAClient } from "../clients/pagopa";
 
 export default class PagoPAClientFactory
   implements IPagoPAClientFactoryInterface {
-  constructor(public readonly pagoPAApiUrl?: string) {}
+  constructor(
+    public readonly pagoPAApiUrl?: string,
+    public readonly pagoPAApiUrlTest?: string
+  ) {}
   /**
    * {@inheritDoc}
    */
-  public getClient(): ReturnType<PagoPAClient> {
-    return PagoPAClient(this.pagoPAApiUrl);
+  public getClient(params?: IGetClientParams): ReturnType<PagoPAClient> {
```

Comment:
I would require just one parameter, that's all you need here:

 ```suggestion
  public getClient(environment: "PRODUCTION" | "TEST"): ReturnType<PagoPAClient> {
```

Reply:
also make it required